packages/api/src/schema.ts

```diff
@@ -1912,6 +1912,93 @@ const schema = gql`
     BAD_REQUEST
   }
 
+  union RulesResult = RulesSuccess | RulesError
+
+  type RulesSuccess {
+    rules: [Rule!]!
+  }
+
+  type Rule {
+    id: ID!
+    name: String
+    query: String!
+    actions: [RuleAction!]!
+    triggerType: RuleTriggerType!
+    createdAt: Date!
+    updatedAt: Date!
+  }
+
+  type RuleAction {
+    type: RuleActionType!
+    value: String
+  }
+
+  enum RuleActionType {
+    ADD_LABEL
+    ARCHIVE
+    MARK_AS_READ
+    SEND_NOTIFICATION
+  }
+
+  enum RuleTriggerType {
+    ON_PAGE_UPDATED
+    CRON
+  }
+
+  type RulesError {
+    errorCodes: [RulesErrorCode!]!
+  }
+
+  enum RulesErrorCode {
+    UNAUTHORIZED
+    BAD_REQUEST
+  }
+
+  input SetRuleInput {
+    id: ID
+    name: String
+    query: String!
+    actions: [RuleActionInput!]!
+    triggerType: RuleTriggerType!
+  }
```

Comment:
@jacksonh `setRule` API can be used to create or update rule based on the existence of `id` field